worker/uniter/runner/context/context.go

```diff
@@ -75,7 +75,6 @@ type Clock interface {
 }
 
 var logger = loggo.GetLogger("juju.worker.uniter.context")
-var mutex = sync.Mutex{}
```

Comment:
I wonder if the intent of this being global is to block concurrent execution of hooks on the same machine. If that's the case then we shouldn't be moving it into the HookContext struct.

@jameinel are you aware of any reasons why this has to be global?